src/client/mapblock_mesh.cpp

```diff
@@ -320,9 +321,14 @@ void final_color_blend(video::SColor *result,
 	b += emphase_blue_when_dark[irr::core::clamp((s32) ((r + g + b) / 3 * 255),
 		0, 255) / 8] / 255.0f;
 
-	result->setRed(core::clamp((s32) (r * 255.0f), 0, 255));
-	result->setGreen(core::clamp((s32) (g * 255.0f), 0, 255));
-	result->setBlue(core::clamp((s32) (b * 255.0f), 0, 255));
+	// Add ambient light
+	r += ambientLight.getRed() / 255.f;
+	g += ambientLight.getGreen() / 255.f;
+	b += ambientLight.getBlue() / 255.f;
```

Comment:
I don't think this is that much of a problem: There are three float divisions and three float additions here. This is not all that much compared to the rest. Additions are very cheap. Divisions are a good bit more expensive. If this is to be optimized, I would suggest just doing the divisions ahead of time rather than introducing branching.

To avoid prematurely optimizing this, I profiled this using `perf`. Apparently, only `0.00049%` of "time" was spent in `final_color_blend`.

Conclusion: Since this is negligible, I see no need to touch this at all.

Reply:
I agree. Thanks for testing.